src/cmd/cmd.cpp

```diff
@@ -290,11 +290,12 @@ void selectiveSyncFixup(OCC::SyncJournalDb *journal, const QStringList &newList)
 
     bool ok = false;
 
-    auto oldBlackListSet = journal->getSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, &ok).toSet();
+    const auto &bl = journal->getSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, &ok);
```

Comment:
Well, OK... that's allowed and the temporary lifetime will be prolonged. That said I'd avoid using that construct it's too obscure for most readers and likely to deduce the wrong thing. The gain from it is really marginal anyway.

(that applies to the whole patch)

Reply:
In this particular case, the call to `getSelectiveSyncList` has to be invoked before the if statement on the next line because it potentially changes the conditional the if statement checks.

Since the `QList::toSet` member function is deprecated, I thought I'd move to the recommended way of constructing the set while I was at it. I can drop that part of the change if you prefer.

I have dropped the extra temporaries from the rest of the patch.

Reply:
I'm slightly confused by your answer here. I wasn't talking about moving the getSelectiveSyncList call but about having bl as const value and not const reference.

Or did I misunderstand what you meant here?

Reply:
Ah, no probably me who misunderstood here. I agree. I'll make it a value.